examples/nextjs-react18-editor-sample/src/components/map-edit-toolbox.tsx

```diff
@@ -0,0 +1,443 @@
+  ViewMode,
+} from "@nebula.gl/edit-modes";
+import { ExportModal, ImportModal } from "@nebula.gl/editor";
+import {
+  ActivityIcon,
+  BoxSelectIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  CircleIcon,
+  DownloadIcon,
+  EraserIcon,
+  HexagonIcon,
+  MapPinIcon,
+  MinusIcon,
+  PlusIcon,
+  PointerIcon,
+  RectangleHorizontalIcon,
+  RedoIcon,
+  RulerIcon,
+  SettingsIcon,
+  ShareIcon,
+  Trash2Icon,
+  TrashIcon,
+  TriangleIcon,
+  UndoIcon,
+  XCircleIcon,
+  XIcon,
+} from "lucide-react";
+import React from "react";
+import styled from "styled-components";
+
+export const SELECTION_TYPE = {
+  NONE: null,
+  RECTANGLE: 'rectangle',
+  POLYGON: 'polygon',
+};
+const Tools = styled.div`
+  position: absolute;
+  display: flex;
+  flex-direction: column;
+  top: 10px;
+  right: 10px;
+`;
+
+const Button = styled.button<{ active?: boolean; kind?: string }>`
+  color: #fff;
+  background: ${({ kind, active }) =>
+    kind === "danger"
+      ? "rgb(180, 40, 40)"
+      : active
+        ? "rgb(0, 105, 217)"
+        : "rgb(90, 98, 94)"};
+  font-size: 1em;
+  font-weight: 400;
+  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
+    "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji",
+    "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
+  border: 1px solid transparent;
+  border-radius: 0.25em;
+  margin: 0.05em;
+  padding: 0.1em 0.2em;
+  :hover {
+    background: rgb(128, 137, 133);
+  }
+`;
+
+const SubToolsContainer = styled.div`
+  position: relative;
+`;
+
+const SubTools = styled.div`
+  display: flex;
+  flex-direction: row-reverse;
+  position: absolute;
+  top: 0;
+  right: 0;
+`;
+
+export type Props = {
+  editMode: any;
+  selectMode: any;
+  selectedFeatureIndexes: any;
+  modeConfig: any;
+  geoJson: any;
+  onSetEditMode: (mode: any) => unknown;
+  onSetSelectMode: (mode: any) => unknown;
+  onSetSelectedFeatureIndexes: (mode: any) => unknown;
+  onSetModeConfig: (modeConfig: any) => unknown;
+  onSetGeoJson: (geojson: any) => unknown;
+  onImport: (imported: any) => unknown;
+};
+const MODE_GROUPS: {
+  modes: {
+    mode: typeof ViewMode;
+    content: JSX.Element;
+    title?: string;
+  }[];
+}[] = [
+    {
+      modes: [{ mode: DrawPointMode, content: <MapPinIcon /> }],
+    },
+    {
+      modes: [
+        {
+          mode: DrawLineStringMode,
+          content: <ActivityIcon />,
+        },
+      ],
+    },
+    {
+      modes: [
+        { mode: DrawPolygonMode, content: <HexagonIcon /> },
+        { mode: DrawRectangleMode, content: <RectangleHorizontalIcon /> },
+        { mode: DrawCircleFromCenterMode, content: <CircleIcon /> },
+      ],
+    },
+    {
+      modes: [
+        { mode: MeasureDistanceMode, content: <RulerIcon /> },
+        { mode: MeasureAngleMode, content: <ChevronLeftIcon /> },
+        { mode: MeasureAreaMode, content: <TriangleIcon /> },
+      ],
+    },
+  ];
+
+function ModeButton({ buttonConfig, mode, onClick }: any) {
```

Comment:
Since this is typescript, should components be type? Should we avoid `any`?

Reply:
I think it should be used as an external example